src/metabase/models/setting.clj

```diff
@@ -98,6 +104,15 @@
   TODO -- we should probably also bind this in sync contexts e.g. functions in [[metabase.sync]]."
   nil)
 
+(def ^:dynamic *user-local-values*
+  "User-local Settings values (as a map of Setting name -> already-deserialized value). This comes from the value of
+  `User.settings` in the application DB. When bound, any Setting that *can* be User-local will have a value from this
+  map returned preferentially to the site-wide value.
+
+  This is normally bound automatically by session middleware, in
+  [[metabase.server.middleware.session/do-with-current-user]]."
+  (atom nil))
```

Comment:
This is an atom so that we can update the value of a setting here when it is written, in conjunction with updating it in the DB. (`*database-local-values*` didn't need to be an atom, at least not yet, since it can't be updated directly)